spec/unit/plugins/windows/dmi_spec.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Author:: Stuart Preston (<stuart@chef.io>)
+# Copyright:: Copyright (c) 2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+describe Ohai::System, "DMI", :windows_only do
+  let(:plugin) { get_plugin("windows/dmi") }
+
+  # Each test case has 3 elements:
+  # * The name of the property as it comes from the Windows APIs
+  # * The transformed snake-case version of the property name
+  # * A unique dummy value per test case
+  CASES = [
+    %w{Depth depth aaa},
+    %w{PartNumber part_number bbb},
+    %w{NumberOfPowerCords number_of_power_cords ccc},
+    %w{SKU sku ddd},
+    %w{SMBIOSAssetTag smbios_asset_tag eee},
+    %w{DeviceID device_id fff},
+    %w{L2CacheSize l2_cache_size ggg},
+  ].freeze
```

Comment:
I originally had this in a tabular format, but chefstyle complained about excessive whitespace. I considered replacing the 3 element array with a Struct or something but thought that was a bit overkill.